src/full/Agda/TypeChecking/Rules/Application.hs

```diff
@@ -927,7 +927,9 @@ checkConstructorApplication cmp org t c args = do
 
   cdef  <- getConInfo c
 
-  checkModality (conName c) cdef
+  ir <- asksTC envInsideReflection
+  let go = if ir then workOnTypes else id
+  go $ checkModality (conName c) cdef
```

Comment:
If I understand correctly this is making all arguments being checked under `workOnTypes` inside macros. But won't this cause metavariables to be created with the wrong quantity? That's the problem I encountered in https://github.com/agda/agda/issues/6400.

Reply:
Yes, the quantity is wrong, and my fix is way too naive, sorry!